sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/db/derived/AllTransactionView.kt

```diff
@@ -0,0 +1,130 @@
+import cash.z.ecc.android.sdk.model.Transaction
+import cash.z.ecc.android.sdk.model.Zatoshi
+import cash.z.ecc.android.sdk.model.ZcashNetwork
+import kotlinx.coroutines.flow.first
+import java.util.Locale
+
+internal class AllTransactionView(
+    private val zcashNetwork: ZcashNetwork,
+    private val sqliteDatabase: SupportSQLiteDatabase
+) {
+    companion object {
+
+        private val ORDER_BY = String.format(
+            Locale.ROOT,
+            "%s DESC, %s DESC",
+            AllTransactionViewDefinition.COLUMN_INTEGER_MINED_HEIGHT,
+            AllTransactionViewDefinition.COLUMN_INTEGER_ID
+        )
+
+        private val SELECTION_BLOCK_RANGE = String.format(
+            Locale.ROOT,
+            "%s >= ? AND %s <= ?", // $NON-NLS
+            AllTransactionViewDefinition.COLUMN_INTEGER_MINED_HEIGHT,
+            AllTransactionViewDefinition.COLUMN_INTEGER_MINED_HEIGHT
+        )
+
+        private val PROJECTION_COUNT = arrayOf("COUNT(*)")
+    }
+
+    private val cursorParser: CursorParser<Transaction> = CursorParser {
+        val idColumnIndex = it.getColumnIndex(AllTransactionViewDefinition.COLUMN_INTEGER_ID)
+        val minedHeightColumnIndex =
+            it.getColumnIndex(AllTransactionViewDefinition.COLUMN_INTEGER_MINED_HEIGHT)
+        val transactionIndexColumnIndex = it.getColumnIndex(
+            AllTransactionViewDefinition
+                .COLUMN_INTEGER_TRANSACTION_INDEX
+        )
+        val rawTransactionIdIndex =
+            it.getColumnIndex(AllTransactionViewDefinition.COLUMN_BLOB_RAW_TRANSACTION_ID)
+        val expiryHeightIndex = it.getColumnIndex(AllTransactionViewDefinition.COLUMN_INTEGER_EXPIRY_HEIGHT)
+        val rawIndex = it.getColumnIndex(AllTransactionViewDefinition.COLUMN_BLOB_RAW)
+        val valueIndex = it.getColumnIndex(AllTransactionViewDefinition.COLUMN_INTEGER_VALUE)
+        val noteCountIndex = it.getColumnIndex(AllTransactionViewDefinition.COLUMN_INT_NOTE_COUNT)
+        val memoCountIndex = it.getColumnIndex(AllTransactionViewDefinition.COLUMN_INTEGER_MEMO_COUNT)
+        // val blockTimeIndex = it.getColumnIndex(AllTransactionViewDefinition.COLUMN_INTEGER_BLOCK_TIME)
+
+        val value = it.getLong(valueIndex)
+        if (value < 0) {
+            Transaction.Sent(
+                id = it.getLong(idColumnIndex),
+                rawId = FirstClassByteArray(it.getBlob(rawTransactionIdIndex)),
+                minedHeight = BlockHeight.new(zcashNetwork, it.getLong(minedHeightColumnIndex)),
+                expiryHeight = BlockHeight.new(zcashNetwork, it.getLong(expiryHeightIndex)),
+                index = it.getLong(transactionIndexColumnIndex),
+                raw = FirstClassByteArray(it.getBlob(rawIndex)),
+                sentTotal = Zatoshi(-value),
+                sentNoteCount = it.getInt(noteCountIndex),
+                memoCount = it.getInt(memoCountIndex),
+                time = 0 // FIXME it.getLong(blockTimeIndex)
```

Comment:
Because I don't have an issue number, ktlint will ensure we don't forget this one before merging.

It needs to get fixed as part of this PR, but is blocked pending https://github.com/zcash/librustzcash/pull/665